aurora/src/pages/index.tsx

```diff
@@ -29,21 +32,57 @@ export default function Home({
         techTypes: new Map<string, string>(techTypes),
     };
 
-    const aboutRef = useRef();
+    const isToastShown: boolean = useAppSelector(
+        (state: RootState) => state.counter.isToastShown
+    );
+    const dispatch = useAppDispatch();
+
+    useEffect(() => {
+        if (isToastShown) {
+            setTimeout(() => {
+                dispatch(changeToastValue());
+            }, 3000);
+        }
+    }, [isToastShown]);
+
+    const aboutRef = useRef<HTMLDivElement>(null)
+    const experienceRef = useRef<HTMLDivElement>(null)
+    const contactRef = useRef<HTMLDivElement>(null)
+    const landingRef = useRef<HTMLDivElement>(null)
+    const projectRef = useRef<HTMLDivElement>(null)
 
     useEffect(() => {
-        const observer = new IntersectionObserver((entries) => {
-            const entry = entries[0];
-            console.log(entry);
-            if (entry.isIntersecting) {
-                console.log('it works');
-            }
-        });
+        const intersectionCallback = (entries :  IntersectionObserverEntry[]) => {
+            entries.forEach((entry) => {
+                if (entry.isIntersecting) {
+                    dispatch(setCurrentPage(entry.target.id))
+                }
+            });
+        }
 
-        if (aboutRef.current) {
-            observer.observe(aboutRef.current);
+        const intersectionOptions = {
+            root: null,
+            rootMargin: "0px",
+            threshold: 0.80
         }
-    }, [aboutRef]);
+
+        const observer = new IntersectionObserver(intersectionCallback, intersectionOptions);
+
+        if (landingRef.current) observer.observe(landingRef.current);
+        if (aboutRef.current) observer.observe(aboutRef.current);
+        if (experienceRef.current) observer.observe(experienceRef.current);
+        if (projectRef.current) observer.observe(projectRef.current);
+        if (contactRef.current) observer.observe(contactRef.current);
+
+        return () => {
+            // ToDo fix potential error, where ref is already changed by the time this function runs
+            if (landingRef.current) observer.unobserve(landingRef.current);
+            if (aboutRef.current) observer.unobserve(aboutRef.current);
+            if (experienceRef.current) observer.unobserve(experienceRef.current);
+            if (projectRef.current) observer.unobserve(projectRef.current);
+            if (contactRef.current) observer.unobserve(contactRef.current);
+        }
+    }, [aboutRef, experienceRef, projectRef, landingRef, contactRef, dispatch]);
```

Comment:
I kinda don't like how for each and everyone you have to be so declarative but on top of my head I also don't have a better solution; needs thought